app/src/main/java/com/example/poste/activities/FolderViewActivity.java

```diff
@@ -72,77 +52,84 @@ protected void onCreate(Bundle savedInstanceState) {
+
+        settingsBut.setOnClickListener(view -> {
+            Intent intent = new Intent(FolderViewActivity.this, EditFolderActivity.class);
+            startActivity(intent);
+        });
+
+        // Create listeners for the post buttons
+        openBut.setOnClickListener(view -> {
+            if (PosteApplication.getSelectedPost() != null) {
+                try {
+                    Intent browserIntent = new Intent(Intent.ACTION_VIEW, Uri.parse(PosteApplication.getSelectedPost().getUrl()));
+                    startActivity(browserIntent);
+                } catch (Exception e) {
+                    e.printStackTrace();
+                    Toast.makeText(FolderViewActivity.this, R.string.internal_error, Toast.LENGTH_LONG).show();
+                }
             }
+        });
 
-            // Fill post view (Recycler View)
-            postRecyclerView.setLayoutManager(new LinearLayoutManager(FolderViewActivity.this));
-            postAdapter = new PostAdapter(
-                    new PostAdapter.ClickListener() {
-                        @Override
-                        public void onItemClick(int position, Post post) {
-                            try {
-                                Intent browserIntent = new Intent(Intent.ACTION_VIEW, Uri.parse(post.getLink()));
-                                startActivity(browserIntent);
-                            } catch (Exception e) {
-                                e.printStackTrace();
-                                Toast.makeText(FolderViewActivity.this, R.string.internal_error, Toast.LENGTH_LONG).show();
-                            }
-                        }
-
-                        @Override
-                        public void onItemLongClick(int position, Post model) {
-//                            Toast.makeText(FolderViewActivity.this, "Share this folder", Toast.LENGTH_LONG).show();
-                        }
-                    },
-                    currentFolder.getPosts()
-            );
-            postRecyclerView.setAdapter(postAdapter);
-            registerForContextMenu(postRecyclerView);
-        } catch (APIException e) {
-            throw new RuntimeException(e);
-        }
-    }
+        editBut.setOnClickListener(view -> {
+            if (PosteApplication.getSelectedPost() != null) {
+                Intent intent = new Intent(FolderViewActivity.this, EditPostActivity.class);
+                startActivity(intent);
+            }
+        });
 
-    @Override
-    public boolean onContextItemSelected(MenuItem item) {
-        Post post = postAdapter.getLocalDataSetItem();
-        Intent intent = null;
-        switch (item.getItemId()) {
-            case R.id.ctx_menu_edit_post:
-                intent = new Intent(FolderViewActivity.this, EditPostActivity.class);
-                intent.putExtra("folderId", currentFolder.getId());
-                intent.putExtra("postId", post.getId());
+        deleteBut.setOnClickListener(view -> {
```

Comment:
Does this button delete a post or delete a folder? It seems to call a method to delete a post, but we're in Folder view, no? If the intent is to delete a post, I would think this should be contained in a PostView activity -- otherwise, which post is this offering to delete for us?

Reply:
See explanation above about the absence of a "PostView".

It would delete whichever post you have selected.
You tap a post from the recycler view, then tap delete. 
Then that post will know who's boss.

Reply:
Roger -- I wasn't familiar with RecyclerView. I had imagined clicking a post would open it, and the options would be a result of a long press or some such. That perhaps is room for future UX improvement (so a user doesn't have to tap twice to open a post -- I seem to recall we had a requirement on number of clicks to get to a post, but don't remember what it is offhand). If what this *actually* does is allow you to highlight a post and then click open / edit / delete, that makes sense. Thank you for the explanation.